lib/idle_task_queue.dart

```diff
@@ -0,0 +1,67 @@
+// @JS()
+library idle_task_queue;
+
+import 'dart:async';
+import 'dart:html';
+import 'dart:js';
+
+import 'package:meta/meta.dart';
+
+/// A handler which will be called during idle time.
+typedef void IdleTaskHandler<T>(T input);
+
+@immutable
+class _IdleTask<T> {
+  final IdleTaskHandler<T> handler;
+  final T input;
+
+  _IdleTask(this.handler, this.input);
+}
+
+/// Adds the task to a queue which will be called during idle time.
+void enqueueIdleTask<T>(IdleTaskHandler<T> handler, T input) {
+  // For unsupported browsers when the polyfill is not available,
+  // we still want the task to be called eventually.
+  if (!_doesBrowserSupportIdleCallback) {
+    new Future.delayed(const Duration()).then((_) => handler(input));
+    return;
+  }
+
+  _taskQueue.add(new _IdleTask<T>(handler, input));
+
+  // Start running the queue if it has not been started.
+  // If there is a task, the queue is already running and doesn't
+  // need to be started again.
+  if (_currentTask == null) {
+    _currentTask = window.requestIdleCallback(
+      allowInterop(_runQueue),
+      <String, dynamic>{'timeout': 1000},
+    );
+  }
+}
+
+void _runQueue(IdleDeadline deadline) {
+  while (!deadline.didTimeout &&
+      deadline.timeRemaining() > 0 &&
+      _taskQueue.isNotEmpty) {
+    final task = _taskQueue.removeAt(0);
+    task.handler(task.input);
+  }
+
+  if (_taskQueue.isNotEmpty) {
+    _currentTask = window.requestIdleCallback(
+      allowInterop(_runQueue),
+      <String, dynamic>{
+        'timeout': 1000,
```

Comment:
should this be configurable?

Reply:
This doesn't need to be configurable. All this means is this is how often we will reset listening for an idle frame. It won't affect the frequency or timing of when tasks are completed.

Reply:
>  It won't affect the frequency or timing of when tasks are completed.
I don't think this is true. From my understanding, this represents the longest time this task can be queued without being called.

So, if there's a lot going on and not a lot of idle periods, this will affect how often these queued tasks are completed.

I'm a little concerned about obfuscating this `timeout` as well as `timeRemaining` from the user (see also comment above about this queue).

It would be nice if the user had the option to configure this timeout, since I wouldn't expect all queued tasks to have the same priority. Perhaps we could have a sane default value that users could then configure.